src/containerapp/azext_containerapp/custom.py

```diff
@@ -2637,3 +2641,240 @@ def list_environment_telemetry_otlp(cmd,
+def execute_session_code_interpreter(cmd,
+                        name,
+                        resource_group_name,
+                        identifier, 
+                        code,
+                        session_pool_location=None,
+                        timeout_in_seconds=None,
+                        ):
+    raw_parameters = locals()
+    session_code_interpreter_decorater = SessionCodeInterpreterCommandsPreviewDecorator(
+        cmd=cmd,
+        client=SessionCodeInterpreterPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    session_code_interpreter_decorater.validate_arguments()
+    session_code_interpreter_decorater.register_provider(CONTAINER_APPS_RP)
+
+    session_code_interpreter_decorater.construct_payload()
+    r = session_code_interpreter_decorater.execute()
+
+    return r
+
+def upload_session_code_interpreter(cmd,
+                        name,
+                        resource_group_name,
+                        identifier, 
+                        filepath
+                        ):
+    raw_parameters = locals()
+    session_code_interpreter_decorater = SessionCodeInterpreterCommandsPreviewDecorator(
+        cmd=cmd,
+        client=SessionCodeInterpreterPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    session_code_interpreter_decorater.register_provider(CONTAINER_APPS_RP)
+    
+    r = session_code_interpreter_decorater.upload()
+
+    return r
+
+def show_file_content_session_code_interpreter(cmd,
+                        name,
+                        resource_group_name,
+                        identifier, 
+                        filename
+                        ):
+    raw_parameters = locals()
+    session_code_interpreter_decorater = SessionCodeInterpreterCommandsPreviewDecorator(
+        cmd=cmd,
+        client=SessionCodeInterpreterPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    session_code_interpreter_decorater.register_provider(CONTAINER_APPS_RP)
+
+    r = session_code_interpreter_decorater.show_file_content()
+
+    return r
+
+def show_file_metadata_session_code_interpreter(cmd,
+                        name,
+                        resource_group_name,
+                        identifier, 
+                        filename
+                        ):
+    raw_parameters = locals()
+    session_code_interpreter_decorater = SessionCodeInterpreterCommandsPreviewDecorator(
+        cmd=cmd,
+        client=SessionCodeInterpreterPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    session_code_interpreter_decorater.register_provider(CONTAINER_APPS_RP)
+
+    r = session_code_interpreter_decorater.show_file_metadata()
+
+    return r
+
+def list_files_session_code_interpreter(cmd,
+                        name,
+                        resource_group_name,
```

Comment:
Looks the `resource_group_name` is not needed, right?

Reply:
In the help it will make customer confused as it is a required argument:
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/31c99aa5-891e-4ec5-afed-6e8173c2c5e0)

Reply:
resource_group_name is needed because we use it to construct the session pool endpoint in get_sessionpool_endpoint which is used by every command https://github.com/Azure/azure-cli-extensions/blob/66ba81b26be1b8992c602ab82b7758748044cba0/src/containerapp/azext_containerapp/containerapp_session_code_interpreter_decorator.py#L54